src/olympia/amo/management/commands/compress_assets.py

```diff
@@ -0,0 +1,237 @@
+import hashlib
+import os
+import re
+import time
+import uuid
+
+import subprocess
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from olympia.lib.jingo_minify_helpers import get_path
+
+
+def path(*args):
+    return os.path.join(settings.STATIC_ROOT, *args)
+
+
+def run_command(command):
+    """Run a command and correctly poll the output and write that to stdout"""
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
+    while True:
+        output = process.stdout.readline()
+        if output == '' and process.poll() is not None:
+            break
+        if output:
+            print(output.strip())
+    rc = process.poll()
+    return rc
+
+
+def ensure_path_exists(path):
+    try:
+        os.makedirs(path)
+    except OSError as e:
+        # If the directory already exists, that is fine. Otherwise re-raise.
+        if e.errno != os.errno.EEXIST:
+            raise
+
+
+class Command(BaseCommand):
+    help = ('Compresses css and js assets defined in settings.MINIFY_BUNDLES')
+    requires_model_validation = False
+
+    checked_hash = {}
+    bundle_hashes = {}
+
+    missing_files = 0
+    minify_skipped = 0
+
+    def generate_build_id(self):
+        return uuid.uuid4().hex[:8]
+
+    def update_hashes(self):
+        # Adds a time based hash on to the build id.
+        self.build_id = '%s-%s' % (self.build_id, hex(int(time.time()))[2:])
+
+        build_id_file = os.path.realpath(
+            os.path.join(settings.ROOT, 'build.py'))
+
+        with open(build_id_file, 'w') as f:
+            f.write('BUILD_ID_CSS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_JS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_IMG = "%s"\n' % self.build_id)
+            f.write('BUNDLE_HASHES = %s\n' % self.bundle_hashes)
+
+    def handle(self, **options):
+        self.build_id = self.generate_build_id()
```

Comment:
nit-ish: We generate `self.build_id` with `generate_build_id` here but only use it in `update_hashes` where we set it again.  Seems like we only need to call `generate_build_id` in `update_hashes`.

Reply:
Indeed, that was left from the refactor of the git integration. Nice find!